packages/dds/tree/docs/storage/compatibility.md

```diff
@@ -0,0 +1,273 @@
+# Compatibility
+
+This document provides concrete guidelines and strategies for organizing code that impacts SharedTree's persisted format.
+It then pivots to the current state of SharedTree's persisted format, and outlines steps to take to align the implementation
+with those guidelines.
+
+## Prerequisites
+
+[This document](../../../SchemaVersioning.md) provides general "best practices" for working with persisted data within the Fluid Framework.
+It's strongly recommended to read through and understand its rationale before continuing with this document,
+as most of the concrete recommendations presented henceforth fall out of those best practices.
+
+## What State is Persisted?
+
+A DDS's persisted format encompasses the format it uses for its summaries as well as its ops (due to [trailing ops](../../../README.md))
+including transitively referenced structured blob data.
+
+Since documents are stored outside of Fluid control (i.e. no type of central data migration is possible),
+DDSes necessarily commit to backwards compatibility of their format for all time.
+
+## Format Management
+
+The actively used format is stored and managed by an `IPersistedConfigStore`.
+This configuration dictates all pieces of policy which impact the set of allowed ops, write format, etc.
```

Comment:
I'd think that information provided at runtime says something across "I can do V1 or V2" (or equivalent)
And information in summary says "I'm V1".
At least that's how we usually make forward compatible changes in schema - ship code dark that supports V2, and then enable it slowly in production (quite often - only for 5% prod, and/or only newly created files), before getting to 100%.

Reply:
Small background: I base my question based on how we historically did forward compatible changes, but without adding a notion of policy (which is not great, but we managed it mostly with assumption that we do it only for one client - Loop, and we can do it relatively safely by timing updates properly, i.e. based on telemetry). Maybe some of the questions do not make any sense (once I understand proposal fully :))

Reply:
> I'd think that information provided at runtime says something across "I can do V1 or V2" (or equivalent) And information in summary says "I'm V1".

That generally aligns with what I'm imagining.

> So config dictates allowed policies, but actual policy is dictated by content of a file, no?

depends what you mean, but there are some important details here. What I ultimately want is:

1. A persisted description of the current policy
2. A commonly-understood (i.e. amongst collaborators) mechanism for changing the current policy
3. Reasonable mechanisms by which DDS authors can declare how aspects of the policy might change over time: some config might be set in stone (e.g. "I loaded some file created before we implemented a super-cool feature and we have absolutely no migration strategy or way to backfill parts of that feature"), whereas others might be changeable (e.g. the SharedTree migration; older documents got converted to the new format once the dark rollout phase was complete). Not sure if there are intermediate settings we'd want here.
4. A way for DDS authors to let application authors control "desired policy" (i.e. the policy that new documents should use, and the policy that old documents should try to convert to within constraints from (3))

(4) is why I've been interested in better support for parameterized DDS factories, and also would let us drop some of the historical assumptions you've mentioned about loop.